ping.go

```diff
@@ -370,15 +372,22 @@ func (p *Pinger) Run() error {
 			}
 		}
 		if p.Count > 0 && p.PacketsRecv >= p.Count {
-			close(p.done)
+			_ = p.Stop()
 			wg.Wait()
 			return nil
 		}
 	}
 }
 
-func (p *Pinger) Stop() {
+func (p *Pinger) Stop() error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if p.torndown {
+		return errors.New("Stop() called more than once")
```

Comment:
Let's define this as a package variable so that it can be tested with `errors.Is(err, ping.ErrTornDown)`.

```
var ErrTornDown = errors.New("Stop() called more than once")
```